datafusion/core/tests/sqllogictests/test_files/interval.slt

```diff
@@ -430,15 +430,11 @@ select '1 month'::interval + '1980-01-01T12:00:00'::timestamp;
 ----
 1980-02-01T12:00:00
 
-query D
+query error DataFusion error: Error during planning: Cannot coerce arithmetic expression Interval\(MonthDayNano\) \- Date32 to valid types
```

Comment:
This query was only supported following #6778 and was done so accidentally - https://github.com/apache/arrow-datafusion/pull/6778#discussion_r1243994954

I tested some other systems, duckdb explicitly doesn't support this, and I couldn't work out how to get mysql to express this, so I think this is fine